parsing/intermediate/dialects/dialect1/parserImpls.ftl

```diff
@@ -967,7 +979,146 @@ SqlCreate SqlCreateTable() :
     {
         return new SqlCreateTableDialect1(s.end(this), createSpecifier, setType,
          volatility, ifNotExists, id, tableAttributes, columnList, query,
-         withData, primaryIndex, indices, onCommitType);
+         withData, primaryIndex, indices, partition, onCommitType);
+    }
+}
+
+SqlTablePartition CreateTablePartitionBy() :
+{
+    final SqlNodeList partitions = new SqlNodeList(getPos());
+    SqlNode e;
+}
+{
+    (
+        e = PartitionExpression()
+        { partitions.add(e); }
+    |
+        <LPAREN>
+        e = PartitionExpression()
+        { partitions.add(e); }
+        (
+            <COMMA>
+            e = PartitionExpression()
+            { partitions.add(e); }
+        )*
+        <RPAREN>
+    )
+    { return new SqlTablePartition(getPos(), partitions); }
+}
+
+SqlNode  PartitionExpression() :
+{
+    final SqlNode e;
+    int constant = 0;
+}
+{
+    (
+        e = RangeN()
+    |
+        e = CaseN()
+    |
+        e = SqlExtractFromDateTime()
+    |
+        e = PartitionByColumnOption()
+    )
+    [
+        <ADD> { constant = UnsignedIntLiteral(); }
+    ]
+    { return new SqlTablePartitionExpression(getPos(), e, constant); }
+}
+
+SqlNode PartitionByColumnOption() :
+{
+    SqlNode e;
+    boolean containAllButSpecifier = false;
```

Comment:
nit: /containAllButSpecifier/containsAllButSpecifier/

Reply:
Fixed in https://github.com/googleinterns/calcite/pull/219/commits/3d2300b6a5d02c53c9c8ca44a8c8550384d6c655